### PR TITLE
Implement Deserialize for Box<str>

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

File: serde/src/de/impls.rs [hunk 1]
```diff
@@ -498,6 +506,14 @@ seq_impl!(
     Vec::with_capacity(visitor.size_hint().0),
     Vec::push);
 
+#[cfg(any(feature = "str", feature = "collections"))]
+impl Deserialize for Box<[T]> where T: Deserialize {
```
Comment (oli-obk) [inline serde/src/de/impls.rs]: `T` needs to be generic

**Comment by nox** [comment]
@oli-obk Oops, fixed.

File: serde/src/de/impls.rs [hunk 2]
```diff
@@ -308,6 +308,14 @@ impl Deserialize for String {
     }
 }
 
+#[cfg(any(feature = "str", feature = "collections"))]
+impl Deserialize for Box<str> {
+    fn deserialize<D>(deserializer: &mut D) -> Result<Self, D::Error> {
```
Comment (oli-obk) [inline serde/src/de/impls.rs]: `D` needs a `Deserializer` bound (also below)

File: serde/src/de/impls.rs [hunk 3]
```diff
@@ -308,6 +308,14 @@ impl Deserialize for String {
...
+        let vec = try!(String::deserialize(deserializer));
```
Comment (oli-obk) [inline serde/src/de/impls.rs]: nit: rename `vec` to `s` or sth appropriate

**Comment by oli-obk** [comment]
@homu r+

**Pull request merged**